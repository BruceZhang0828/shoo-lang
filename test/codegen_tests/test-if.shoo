string x = "Hello "+"World"+"!";
string y = "what";
if (x!=y) {
	println("GOOD");
} else {
	println("SHOULDNT HAVE PRINTED THIS");
}

int i = 0;
int j = 0;
if(i==j) {
	println("JOB");
}

if (i > 1) {
    println("no");
} elif (i == 0) {
    println("yes");
} else {
    println("none");
}

if (i == 0) {

} else {

}


if (i == 4) {
    println(str_of_int(4));
} elif (i == 3) {
    println(str_of_int(3));
} elif (i == 2) {
    println(str_of_int(2));
} elif (i == 1) {
    println(str_of_int(1));
} elif (i == 0) {
    println(str_of_int(0));
} else {
    println("what??");
}
