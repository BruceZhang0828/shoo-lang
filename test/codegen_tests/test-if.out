GOOD
JOB
yes
0
